pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -491,6 +492,14 @@ func (c *Controller) preparePacket(tf *crdv1beta1.Traceflow, intf *interfacestor
 		} else if packet.DestinationIP.To4() != nil {
 			return nil, errors.New("destination Service does not have an IPv6 ClusterIP")
 		}
+		switch dstSvc.Spec.Ports[0].Protocol {
+		case corev1.ProtocolTCP:
+			packet.IPProto = protocol.Type_TCP
+			packet.TCPFlags = uint8(2)
+		case corev1.ProtocolUDP:
+			packet.IPProto = protocol.Type_UDP
+		}
+		packet.DestinationPort = uint16(dstSvc.Spec.Ports[0].Port)
```

Comment:
I assume these are just defaults and that fields in `tf.Spec.Packet` can still override them?

Reply:
default fields will be overridden only when `tf.Spec.Packet.TransportHeader.TCP` or `tf.Spec.Packet.TransportHeader.UDP` is provided.